test/functorch/test_dispatch.py

```diff
@@ -0,0 +1,135 @@
+# Owner(s): ["module: functorch"]
+from functools import wraps
+import unittest
+
+from torch.testing._internal.common_utils import (
+    _TestParametrizer,
+    TestCase,
+    run_tests,
+    instantiate_parametrized_tests,
+)
+
+from torch._C import (
+    _dispatch_get_registrations_for_dispatch_key as get_registrations_for_dispatch_key,
+)
+
+xfail_functorch_batched = {
+    "aten::argsort",
+    "aten::chunk",
+    "aten::contiguous",
+    "aten::cross",
+    "aten::diag",
+    "aten::flatten.using_ints",
+    "aten::floor_divide.Scalar",
+    "aten::gather_backward",
+    "aten::imag",
+    "aten::is_nonzero",
+    "aten::isfinite",
+    "aten::isreal",
+    "aten::item",
+    "aten::linalg_matrix_power",
+    "aten::linalg_matrix_rank.atol_rtol_float",
+    "aten::linalg_matrix_rank.atol_rtol_tensor",
+    "aten::linalg_pinv",
+    "aten::linalg_pinv.atol_rtol_float",
+    "aten::linalg_slogdet",
+    "aten::linear",
+    "aten::log_sigmoid",
+    "aten::log_softmax.int",
+    "aten::logdet",
+    "aten::masked_select_backward",
+    "aten::movedim.intlist",
+    "aten::one_hot",
+    "aten::real",
+    "aten::relu6",
+    "aten::relu6_",
+    "aten::selu",
+    "aten::selu_",
+    "aten::silu_backward",
+    "aten::special_digamma",
+    "aten::special_erf",
+    "aten::special_erfc",
+    "aten::special_erfinv",
+    "aten::special_exp2",
+    "aten::special_expit",
+    "aten::special_expm1",
+    "aten::special_gammaln",
+    "aten::special_i0",
+    "aten::special_log1p",
+    "aten::special_ndtr",
+    "aten::special_psi",
+    "aten::special_round",
+    "aten::special_sinc",
+    "aten::special_xlogy",
+    "aten::special_xlogy.other_scalar",
+    "aten::special_xlogy.self_scalar",
+    "aten::tensor_split.indices",
+    "aten::tensor_split.sections",
+    "aten::to.device",
+    "aten::to.dtype",
+    "aten::to.dtype_layout",
+    "aten::to.other",
+    "aten::upsample_bicubic2d.vec",
+    "aten::upsample_bilinear2d.vec",
+    "aten::upsample_linear1d.vec",
+    "aten::upsample_nearest1d.vec",
+    "aten::upsample_nearest2d.vec",
+    "aten::upsample_nearest3d.vec",
+    "aten::upsample_trilinear3d.vec",
+    "aten::where",
+}
+
+xfail_functorch_batched_decomposition = {
+    'aten::diagonal_copy',
+    'aten::is_same_size',
+    'aten::t',
+    'aten::t_',
+    'aten::unfold_copy',
+}
+
+class dispatch_registrations(_TestParametrizer):
```

Comment:
I kept this class in the test file but happy to move it out if it makse sense.

Reply:
Here seems fine for now, if someone else needs it they can move it out